### PR TITLE
ENH: Create extension point for data portals

### PR DESCRIPTION
**Pull request by jnazaren**
Creates an extension point for `DataPortal`, which currently defaults to `HistoricDataPortal`, the renamed version of `DataPortal`. This PR follows the method used in #2210.

### CODE REVIEW

File: zipline/utils/run_algo.py [hunk 1]
```diff
@@ -179,6 +171,24 @@ def choose_loader(column):
         except ValueError as e:
             raise _RunAlgoError(str(e))
 
+    data = HistoricDataPortal(
```
Comment (richafrank) [inline zipline/utils/run_algo.py]: Would it make sense to create this only in an `else` branch below? A bunch of branches below ignore this value.

File: tests/test_algorithm.py [hunk 2]
```diff
@@ -4301,7 +4301,7 @@ def make_equity_info(cls):
             orient='index',
         )
 
-    # XXX: This suite doesn't use the data in its DataPortal; it uses a
+    # XXX: This suite doesn't use the data in its HistoricDataPortal; it uses a
```
Comment (richafrank) [inline tests/test_algorithm.py]: Is this talking specifically about the implementation, or the interface?

**Comment by richafrank** [comment]
Will be curious what @ehebert thinks of this class hierarchy being a public interface...

**Comment by richafrank** [comment]
Do you expect sibling classes to the HistoricalDataPortal, or will they be subclasses? Makes me wonder if Historical is the right name.

**Comment by jnazaren** [comment]
@richafrank I agree - I called it HistoricDataPorta because it made sense in terms of the type of data it was providing, but since I do imagine there would be subclasses to this implementation in the future, there's probably a better name for it. Would BasicDataPortal or DefaultDataPortal be more accurate?

**Comment by jnazaren** [comment]
Closing in favor of #2269, which opens up the possibility of making readers an extension point instead.

**Pull request closed without merging**